src/simulation/elements/SLCN.cpp

```diff
@@ -108,19 +108,19 @@ static int update(UPDATE_FUNC_ARGS)
 
 static int graphics(GRAPHICS_FUNC_ARGS)
 {
-	int curr_colour = SLCN_COLOUR[(cpart->tmp >> 12) & 15];
+	auto curr_colour = RGB<uint8_t>::Unpack(SLCN_COLOUR[(cpart->tmp >> 12) & 15]);
 	if (cpart->tmp & 0x800) // mix with next colour if phase is at least halfway there
 	{
-		int next_colour = SLCN_COLOUR[(cpart->tmp >> 16) & 15];
-		curr_colour = PIXRGB(
-			(PIXR(curr_colour) + PIXR(next_colour)) / 2,
-			(PIXG(curr_colour) + PIXG(next_colour)) / 2,
-			(PIXB(curr_colour) + PIXB(next_colour)) / 2
+		auto next_colour = RGB<uint8_t>::Unpack(SLCN_COLOUR[(cpart->tmp >> 16) & 15]);
+		curr_colour = RGB<uint8_t>(
+			(curr_colour.Red   + next_colour.Red) / 2,
```

Comment:
I wonder if Blend with a constant alpha of 0x80 or 0x7F would optimize to the right thing here. (Dont do it if not)